File: Remove-Duplicates-from-Sorted-Array/Remove-Duplicates-from-Sorted-Array.py


class Solution:
    # @param a list of integers
    # @return an integer
    def removeDuplicates(self, A):
        if len(A)==0:
            return 0
        last=A[0]
        pos=0
        for i in range(len(A)):
            if last!=A[i]:
                A[pos]=last
                pos+=1
                last=A[i]
        A[pos]=last
        pos+=1
        return pos
 

File: Remove-Duplicates-from-Sorted-Array/Remove-Duplicates-from-Sorted-Array.cpp

class Solution {
public:
    int removeDuplicates(int A[], int n) {
        int ans=1;
        for(int i=1;i<n;i++)
            if(A[i]!=A[ans-1])
                A[ans++]=A[i];
        return n==0?0:ans;
    }
};


 